lib/fs/fs.ml

```diff
@@ -15,8 +22,8 @@ type cursor =
 
 (* Extracts the file name from a tree node *)
 let file_name = function
-  | File (name, _, _) -> name
-  | Dir (name, _) -> name
+  | File { name; _ } -> name
+  | Dir { name; _ } -> name
```

Comment:
This was actually the main motivation behind doing this minor refactoring.

The internal contents of constructors like `File` and `Dir` might change by adding and removing some fields (I already expect a few possible changes). But these changes will be decoupled from functions that don't affect them, so it'll be easier to perform them.

Even such minor things matter when a project grows.